addons/account/wizard/account_payment_register.py

```diff
@@ -380,7 +380,8 @@ def _compute_currency_id(self):
     @api.depends('payment_type', 'company_id', 'can_edit_wizard')
     def _compute_available_journal_ids(self):
         for wizard in self:
-            if wizard.can_edit_wizard:
+            single_payment_type = len(set(batch['payment_values']['payment_type'] for batch in wizard._get_batches())) == 1
+            if wizard.can_edit_wizard or single_payment_type:
                 batch = wizard._get_batches()[0]
                 wizard.available_journal_ids = wizard._get_batch_available_journals(batch)
```

Comment:
Technically, it's working but I really don't like calling that part with the first batch you find. Maybe it's more logical to call `_get_batch_available_journals` for each batch and aggregate all the results together instead